### PR TITLE
Support class based flex flow with yaml in chat ui

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description
- pf flow test --ui --flow flow.flex.yaml

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_sdk/_orchestrator/experiment_orchestrator.py [hunk 1]
```diff
@@ -852,6 +856,7 @@ def __init__(
         self.node_inputs = {}  # E.g. {'main': {'url': 'https://abc'}}
         self.test_data = ExperimentHelper.prepare_test_data(inputs, template)
         self.test_inputs = {input.name: input.default for input in template.inputs}
+        self.init = init
```
Comment (YingChen1996) [inline src/promptflow-devkit/promptflow/_sdk/_orchestrator/experiment_orchestrator.py:859]: Not sure whether add init here is proper

File: src/promptflow/tests/test_configs/experiments/class-based-eager-flow-exp-template/flow.exp.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  - name: my_data
+    path: ./inputs.jsonl
+
+nodes:
+  - name: main
+    type: flow
+    path: ../../eager_flows/basic_callable_class/flow.flex.yaml
+    inputs:
+      func_input: ${data.my_data.func_input}
+
```
Comment (D-W-) [inline src/promptflow/tests/test_configs/experiments/class-based-eager-flow-exp-template/flow.exp.yaml:15]: Add a `init` here?

Comment (YingChen1996) [reply src/promptflow/tests/test_configs/experiments/class-based-eager-flow-exp-template/flow.exp.yaml:18]: added

File: src/promptflow-devkit/promptflow/_sdk/_service/apis/connection.py [hunk 3]
```diff
@@ -77,10 +77,6 @@ def _get_connection_operation(working_directory=None):
 class ConnectionList(Resource):
     @api.doc(parser=working_directory_parser, description="List all connection")
     @api.marshal_with(list_connection_field, skip_none=True, as_list=True)
-    @local_user_only
```
Comment (YingChen1996) [inline src/promptflow-devkit/promptflow/_sdk/_service/apis/connection.py:80]: @lalala123123  to double-confirm here

File: src/promptflow-devkit/promptflow/_sdk/_orchestrator/experiment_orchestrator.py [hunk 4]
```diff
@@ -762,10 +767,10 @@ def __init__(self, template: ExperimentTemplate, environment_variables=None, ses
         # Generate line run id for node
         timestamp = datetime.now().strftime("%Y%m%d_%H%M%S_%f")
         self.node_name_to_id = {node.name: f"{node.name}_attempt{timestamp}" for node in template.nodes}
-        skip_node_name = kwargs.get("skip_node_name", None)
+        context_node_name = kwargs.get("context_node_name", None)
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_orchestrator/experiment_orchestrator.py:770]: add some comments on what's a context_node

File: src/promptflow-devkit/promptflow/_sdk/_service/apis/experiment.py [hunk 5]
```diff
@@ -72,22 +82,26 @@ def post(self):
...
             flow_path_dir, flow_path_file = resolve_flow_path(override_flow_path)
             override_flow_path = (flow_path_dir / flow_path_file).as_posix()
-            context = {"inputs": inputs, "node": override_flow_path}
+            context = {"inputs": inputs, "node": override_flow_path, "run_id": main_flow_run_id, "init": init}
+
         if output_path is None:
             filename = str(uuid.uuid4())
             if os.path.isdir(experiment_template):
-                output_path = Path(experiment_template) / filename
+                output_path = Path(experiment_template) / PROMPT_FLOW_DIR_NAME / filename
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_service/apis/experiment.py:102]: looks like we have some code can be shared on outputs path logic

File: src/promptflow-devkit/promptflow/_sdk/_service/apis/connection.py (hunk 3 above)
Comment (elliotzh) [reply src/promptflow-devkit/promptflow/_sdk/_service/apis/connection.py:80]: why we remove the local user only tag?
If this is for extension, they can set an extra header in request to pass this check

File: src/promptflow-devkit/promptflow/_sdk/_service/app.py [hunk 6]
```diff
@@ -129,7 +122,7 @@ def handle_exception(e):
             formatted_exception = FormattedException(e)
 
             def handle_windows_path(obj):
-                if isinstance(obj, WindowsPath):
+                if isinstance(obj, (WindowsPath, PosixPath)):
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_service/app.py:125]: Can we check on the base class?

File: src/promptflow-devkit/promptflow/_sdk/operations/_experiment_operations.py [hunk 7]
```diff
@@ -170,9 +170,14 @@ def test(self, experiment: Experiment, inputs=None, environment_variables=None,
         experiment_template = _load_experiment_template(experiment)
         output_path = kwargs.pop("output_path", None)
         session = kwargs.pop("session", None)
-
         return ExperimentOrchestrator(client=self._client, experiment=None).test(
-            experiment_template, None, inputs, environment_variables, output_path=output_path, session=session, **kwargs
+            experiment_template,
+            None,
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/operations/_experiment_operations.py:175]: what's None here? give it a key

File: src/promptflow-devkit/promptflow/_sdk/_service/apis/connection.py (hunk 3 above)
Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_sdk/_service/apis/connection.py:80]: offline sycned, only keep header in ConnectionWithSecret

**Review by wangchao1230 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Review by elliotzh — APPROVED** [review]

**Pull request merged**